src/components/operations/operation-details/react/runtime/OperationConsole.tsx

```diff
@@ -0,0 +1,319 @@
+    routeHelper: RouteHelper;
+    settingsProvider: ISettingsProvider;
+    sessionManager: SessionManager;
+    httpClient: HttpClient;
+}
+
+interface StoredCredentials {
+    grantType: string;
+    accessToken: string;
+}
+
+interface OAuthSession {
+    [apiName: string]: StoredCredentials;
+}
+
+export const OperationConsole = ({
+    isOpen,
+    setIsOpen,
+    api,
+    operation,
+    hostnames,
+    useCorsProxy,
+    apiService,
+    usersService,
+    productService,
+    oauthService,
+    tenantService,
+    routeHelper,
+    settingsProvider,
+    sessionManager,
+    httpClient
+}: OperationConsoleProps) => {
+    const [working, setWorking] = useState<boolean>(false);
+    const [authorizationServers, setAuthorizationServers] = useState<AuthorizationServer[]>([]);
+    const [products, setProducts] = useState<any[]>([]);
+    const [selectedSubscriptionKey, setSelectedSubscriptionKey] = useState<string>(null);
+    const [isConsumptionMode, setIsConsumptionMode] = useState<boolean>(false);
+    const [backendUrl, setBackendUrl] = useState<string>("");
+
+    const consoleOperation = useRef(new ConsoleOperation(api, operation));
+    const [forceRerender, setForceRerender] = useState<number>(1);
+    const rerender = useCallback(() => setForceRerender(old => old + 1), []);
+
+    useEffect(() => {
+        setWorking(true);
+        consoleOperation.current.host.hostname(hostnames[0]);        
+        Promise.all([            
+            getAuthServers(api, oauthService).then(authServers => {
+                setAuthorizationServers(authServers);
+                if (authServers.length > 0) {
+                    setupOAuth(api, authServers[0], consoleOperation.current.request.headers(), sessionManager).then(newHeaders => {
+                        consoleOperation.current.request.headers(newHeaders);
+                    });
+                }
+            }),
+            loadSubscriptionKeys(api, apiService, productService, usersService).then(products => {
+                setProducts(products);
+                if (products.length > 0) {
+                    setSelectedSubscriptionKey(products[0].subscriptionKeys[0]?.value);
+                }
+            }),
+            getSkuName().then(skuName => setIsConsumptionMode(skuName === ServiceSkuName.Consumption)),
+            getBackendUrl(settingsProvider).then(url => setBackendUrl(url))
+        ])
+        .catch(error => new Error(`Unable to load the console details. Error: ${error.message}`))
+        .finally(() => {
+            setWorking(false);
+        });
+    }, [api, operation, consoleOperation]);
+
+    useEffect(() => {
+        !isConsumptionMode && api.type !== TypeOfApi.webSocket
+            && !consoleOperation.current.request.headers().some(header => header.name() === KnownHttpHeaders.CacheControl)
+            && consoleOperation.current.setHeader(KnownHttpHeaders.CacheControl, "no-cache", "string", "Disable caching.");
```

Comment:
I'd put this to "normal" 
if (...) { ... }
block or is there some reason it's in this "functional" format?